sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -180,18 +212,248 @@ object Hash {
       val ss = set.toSeq.sorted[String]
       iterateOver(ss.iterator, ss.size)(_ addString _)
     }
+
+    final def addOptional[S](opt: Option[S], hashS: this.type => S => this.type): this.type = {
+      opt match {
+        case None => addInt(0)
+        case Some(value) => hashS(addInt(1))(value)
```

Comment:
`addInt` will insert `0 (int)` or `1 (int)` into the context. Wouldn't it be better to add `None` / `Some` into the context?